frb_codegen/src/config/opts_parser.rs

```diff
@@ -1,27 +1,32 @@
 use crate::config::opts::Opts;
 use crate::config::raw_opts::RawOpts;
 use crate::config::refine_c_output::get_refined_c_output;
-use crate::utils::misc::{find_all_duplicates, BlockIndex};
+use crate::utils::misc::{
+    get_symbols_if_no_duplicates, is_multi_blocks_case, is_same_directory, BlockIndex,
+    ExtraTraitForVec, PathExt,
+};
 use anyhow::*;
 use clap::CommandFactory;
 use convert_case::{Case, Casing};
 use itertools::Itertools;
 use std::borrow::Cow;
+
 use std::ffi::OsString;
 use std::fs::File;
 use std::path::{Path, PathBuf};
 use std::str::FromStr;
 use std::{env, fs};
 use toml::Value;
 
-pub fn config_parse(mut raw: RawOpts) -> Vec<Opts> {
+pub fn config_parse(mut raw: RawOpts) -> Result<(Vec<Opts>, Vec<String>)> {
```

Comment:
nit: I am not sure... `config_parse` seems to return all configs *plus* all symbols. But the name is "config_parse", meaning it wants to "parse the config", which seems not to be related with symbols.

If we want to have those symbols, maybe we can create a new parallel function, say `parse_all_symbols`, etc

Reply:
Since the config and symbols  are parsed always in pair, here I simply refine the `config_parse` to `parse_config_and_symbols` with a doc:
``` rust
/// Parses the raw command-line options,
/// and returns a tuple containing a vector of `Opts` structs
/// and a vector of unique symbol name(s)/Api(s).
/// Both the 2 are essential for code generation.
pub fn parse_configs_and_symbols(mut raw: RawOpts) -> Result<(Vec<Opts>, Vec<String>)> {
...
}
```
I think this is enough at present.  2f868da